nums/core/array/blockarray.py

```diff
@@ -125,12 +125,14 @@ def touch(self):
         self.system.get(oids)
         return self
 
-    def reshape(self, shape=None, **kwargs):
+    def reshape(self, *shape, **kwargs):
         block_shape = kwargs.get("block_shape", None)
         if array_utils.is_int(shape):
             shape = (shape,)
-        elif shape is None:
+        elif len(shape) == 0:
             shape = self.shape
+        elif len(shape) == 1 and isinstance(shape[0], (tuple, list)):
```

Comment:
Can you update this to be:

```
if isinstance(shape[0], (tuple, list)):
    assert len(shape) == 1
    ...
```

I'm assuming you cannot pass a list of lists or list of tuples, but please confirm by checking NumPy behavior. Also, can you add a final `else` if none of the above conditions are satisfied, checking that the `shape` provided is a list of type int?

Reply:
Will do. I can confirm you can't pass in a list of lists/tuples.